pkg/auth/ldaputil/client.go

```diff
@@ -78,3 +84,20 @@ func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
 		return nil, fmt.Errorf("unsupported scheme %q", l.Scheme)
 	}
 }
+
+// ConnectAndBind returns an established LDAP connection to which a bind has occurred if a bind DN and
+// password were given
+func (l *LDAPClientConfig) Connect() (*ldap.Conn, error) {
+	connection, err := l.CreateConnection()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(l.BindDN) > 0 {
+		if err := connection.Bind(l.BindDN, l.BindPassword); err != nil {
+			return nil, err
```

Comment:
Good catch. If you defer a `Close()` on the bound connection returned from this method, however, that will correctly close the connection? I have no qualms about breaking the bind into a separate method, as long as closing the bound connection will work as expected.